app-ios/Sources/SearchFeature/FavoriteToggleButtonView.swift

```diff
@@ -0,0 +1,54 @@
+import appioscombined
+import SwiftUI
+import Theme
+
+struct FavoriteToggleButtonView: View {
+    let isFavorite: Bool
+    let onFavoriteToggle: () -> Void
+    var body: some View {
+        Button {
+            onFavoriteToggle()
+        } label: {
+            HStack(spacing: 12) {
+                if isFavorite {
+                    Image(systemName: "checkmark")
+                        .resizable()
+                        .frame(width: 8, height: 8)
+                        .foregroundColor(isFavorite ? AssetColors.onSurface.swiftUIColor :
+                                         AssetColors.white.swiftUIColor)
+                }
+                Text(StringsKt.shared.search_filter_favorites.localized())
+                    .foregroundColor(isFavorite ? AssetColors.onSurface.swiftUIColor : AssetColors.white.swiftUIColor)
```

Comment:
I think following colors are correct. Please check that.

isFavorite|Color
--|--
true| onSecondaryContainer
false| onSurfaceVariant

Reply:
And how about specify this `foregroundColor` in `HStack`?
In that case, `Image` and `Text` receive the same designation.